2016-04-08-rules-and-queries.md

```diff
@@ -0,0 +1,342 @@
+                "DELIVERY_DATE": [{"value": "0"}],
+                "IMGURL": [{"value": "www.image1.com"}, {"value": "www.image2.com"}]
+                "GIFTS": [{
+                	"attributes": {"ID":[{"value": "KVP-12"}]}, 
+                	"children": {"GIFT":[{"value": "Free Delivery"}]}
+                }]
+                
+            },
+            "metadata": {
+                "..."
+            }
+        },
+        {
+            "id": "234",
+            "created_at": "2016-01-07T13:56:42+00:00",
+            "updated_at": "2016-04-28T17:18:50+00:00",
+            "output_changed_at": null,
+            "data": {
+                "ITEM_ID": [{"value": "5336"}],
+                "PRODUCTNAME": [{"value": "Epson LX 100 EPS"}],
+                "DELIVERY_DATE": [{"value": "3"}],
+                "EAN": [{"value": "0010313601581"}],
+                "IMGURL": [{"value": "www.image4.com"}, {"value": "www.image5.com"}]
+                "GIFTS": [{
+                	"attributes": {"ID":[{"value": "KVP-175"}]}, 
+                	"children": {"GIFT":[{"value": "Discount 10$"}]}
+                }]
+            },
+            "metadata": {
+                "..."
+            }
+        }
+    ]
+}
+```
+
+Explanation of the fields:
+
+* `project_id` - ID of the project. Changes in products by rules are applied to this project.
+* `rule_id` - ID of the rule in Mergado, each rule is represented by one row in the UI.
+* `apply_log_id` - ID of the rule application's log.
+* `request_id` - ID of the request different for each request. Note that the ID doesn't change when the request is retried (see Retrying on Errors bellow).
+* `current_format` - Name of the format in which the products are represented when this rule is applied.
+* `data` - A list of products.
+    + `id` - ID of the product.
+    + `created_at` - When the product has been imported in Mergado.
+    + `updated_at` - The last time the product changed.
+    + `output_changed_at` - The last time the product changed its output values.
+    + `data` - Nested structure representing product data. These values are altered by rules with higher priority (applied sooner in the chain of rules).
+    + `metadata` - Cached data for the whole application procces that you send before in response for this particular product in another rule.
+
+The request is considered to be a success if the server replies with a 200 OK HTTP status code and the body of the response contains products’ data in the same format. The application is not required to return all products, it is required to return only the products and subtrees with top level elements as roots that were processed and should be changed in some way. If application wants to change the values ​​of elements, it is necessary to send the entire subtree in the response, starting with the top level element, in its final form. For example, the server's response might be:
```

Comment:
ve verzi `until` máš 200 OK takhle: `200 OK` (mezi apostrofy, nebo co to je za znak), tak zachovej konzistenci

Reply:
opravene